doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,229 @@
+
+   For MSI it is the product code.  Typically a GUID that is typically found in the uninstall registry location and includes the brackets.
+   For example: ```{5740BD44-B58D-321A-AFC0-6D3D4556DD6C}```
+
+   ```HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+    
+   ```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+
+   For inno, wix, nullsoft, and exe, the SystemAppId should be a string that is located in either of the Uninstall keys above. 
+
+* *Zip files*
+We expect to add Zip Support.  As such, ZIP does mot have a unique identifier.  The Windows Package Manager will need to track where the files were installed, so that it can remove the file and preserve the users files.
+
+### Upgrade
+
+The **upgrade** command is designed to update one or more applications.  The **upgrade** command when executed with no parameters or with --all will update all packages that have updates available.
+
+
+## Usage
+
+```winget upgrade [[-q] <query>] [<options>]```
+
+
+
+**Upgrade** when executed by itself will upgrade all apps ready for updates by the Windows Package Manager.
+
+
+| Argument  | Description |
+|--------------|-------------|
+| **-q,--query** |  The query used to search for an application. |
+| **-?, --help** |  Gets additional help on this command. |
+
+## Options
+
+The following options are available.
+
+| Option  | Description |
+|--------------|-------------|
+| **--all**         |  update all applications that are installed and have updates available.|
+| **--id**         |  Filter results by ID. |
+| **--name**   |      Filter results by name. |
+| **--moniker**   |  Filter results by application moniker. |
+| **--tag** |     Filter results by tag
+| **--command**  | Filter results by command
+| **-s,--source** |   Find the application using the specified [source](source.md). |
+| **-e,--exact**     | Find the application using exact match. |
+
+
+*Upgrade app when no update is available*  
+If the user attempts to update an app that there is no known update for, the package manager will issue an error.
+"There is no update available from the community repository.  If you need to reinstall or repair the app, try ```winget repair <appname>```.  
+
+*Dependencies*
+Handling dependencies will be a challenge once Windows Package Manager supports dependencies.  Dependencies are not covered as part of this spec.
+
+**Open issue** if the application is not side by side, it will force an uninstall of the previous version to upgrade.  we will need to assume the installer wil handle.
+
+### Uninstall
+
+The **uninstall** command is designed to uninstall an application.   
+
+## Usage
+
+```winget uninstall [[-q] <query>] [<options>]```
```

Comment:
Does uninstall handle apps installed outside of WinGet?
If there're multiple matches with the query. Should we return a warning message saying "please refine the input ..." like what install command does today?